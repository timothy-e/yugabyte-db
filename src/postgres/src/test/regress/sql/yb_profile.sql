--
-- YB_LOGIN_PROFILE Testsuite: Testing Statments for LOGIN PROFILES.
--
\h CREATE PROFILE
\h DROP PROFILE

--
-- pg_catalog alterations. Validate columns of pg_yb_profile and oids.
--
\d pg_yb_profile
SELECT oid, relname, reltype, relnatts FROM pg_class WHERE relname IN ('pg_yb_profile', 'pg_yb_profile_oid_index');
SELECT oid, typname, typrelid FROM pg_type WHERE typname LIKE 'pg_yb_profile';
SELECT pg_describe_object('pg_yb_profile'::regclass::oid, oid, 0) FROM pg_yb_profile;

--
-- CREATE PROFILE
--
SELECT oid, prfname, prffailedloginattempts FROM pg_catalog.pg_yb_profile ORDER BY oid;

CREATE PROFILE test_profile FAILED ATTEMPTS 3;

SELECT prfname, prffailedloginattempts FROM pg_catalog.pg_yb_profile ORDER BY OID;

--Fail because it is a duplicate name

CREATE PROFILE test_profile FAILED ATTEMPTS 4;

--
-- DROP PROFILE
--

DROP PROFILE test_profile;

-- fail: does not exist
DROP PROFILE test_profile;

-- fail: cannot delete default profile
DROP PROFILE "default";
