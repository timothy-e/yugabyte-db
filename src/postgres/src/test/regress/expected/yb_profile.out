--
-- YB_PROFILE Testsuite: Testing statements for profiles.
--
--
-- pg_catalog alterations. Validate columns of pg_yb_profile and oids.
--
\d pg_yb_profile
                   Table "pg_catalog.pg_yb_profile"
          Column           |  Type   | Collation | Nullable | Default
---------------------------+---------+-----------+----------+---------
 prfname                   | name    |           | not null |
 prfmaxfailedloginattempts | integer |           | not null |
 prfpasswordlocktime       | integer |           | not null |
Indexes:
    "pg_yb_profile_oid_index" PRIMARY KEY, lsm (oid ASC), tablespace "pg_global"
    "pg_yb_profile_prfname_index" UNIQUE, lsm (prfname ASC), tablespace "pg_global"
Tablespace: "pg_global"

SELECT oid, relname, reltype, relnatts FROM pg_class WHERE relname IN ('pg_yb_profile', 'pg_yb_profile_oid_index');
 oid  |         relname         | reltype | relnatts
------+-------------------------+---------+----------
 8051 | pg_yb_profile           |    8053 |        3
 8052 | pg_yb_profile_oid_index |       0 |        1
(2 rows)

SELECT oid, typname, typrelid FROM pg_type WHERE typname LIKE 'pg_yb_profile';
 oid  |    typname    | typrelid
------+---------------+----------
 8053 | pg_yb_profile |     8051
(1 row)

SELECT pg_describe_object('pg_yb_profile'::regclass::oid, oid, 0) FROM pg_yb_profile;
 pg_describe_object
--------------------
(0 rows)

--
-- CREATE PROFILE
--
SELECT oid, prfname, prfmaxfailedloginattempts FROM pg_catalog.pg_yb_profile ORDER BY oid;
 oid | prfname | prfmaxfailedloginattempts
-----+---------+---------------------------
(0 rows)

CREATE PROFILE test_profile LIMIT FAILED_LOGIN_ATTEMPTS 3;
SELECT prfname, prfmaxfailedloginattempts FROM pg_catalog.pg_yb_profile ORDER BY OID;
   prfname    | prfmaxfailedloginattempts
--------------+---------------------------
 test_profile |                         3
(1 row)

-- Fail because it is a duplicate name
CREATE PROFILE test_profile LIMIT FAILED_LOGIN_ATTEMPTS 4;
ERROR:  profile "test_profile" already exists
-- Fail because -ve numbers are not allowed
CREATE PROFILE test_profile LIMIT FAILED_LOGIN_ATTEMPTS -1;
ERROR:  syntax error at or near "-"
LINE 1: CREATE PROFILE test_profile LIMIT FAILED_LOGIN_ATTEMPTS -1;
                                                                ^
--
-- DROP PROFILE
--
DROP PROFILE test_profile;
-- fail: does not exist
DROP PROFILE test_profile;
ERROR:  profile "test_profile" does not exist
--
-- Test IF EXISTS clause
--
DROP PROFILE IF EXISTS non_existing;
NOTICE:  profile "non_existing" does not exist, skipping
CREATE PROFILE exists_profile LIMIT FAILED_LOGIN_ATTEMPTS 3;
DROP PROFILE IF EXISTS exists_profile;
